addon/store.js

```diff
@@ -174,17 +179,56 @@ var Store = Ember.Object.extend({
       }
       applyHeaders(opt.headers, newHeaders, true);
 
-      return self.request({
+      // push all promises in arrray for 1 url
+      queuedRequest = {
         url: url,
-        depaginate: opt.depaginate,
-        headers: newHeaders,
-      }).then(function(result) {
-        if ( isForAll )
-        {
-          self.get('_foundAll').set(type,true);
-        }
-        return result;
-      });
+        promises: null,
+      };
+
+      var later;
+
+      // check to see if the request is in the promiseQueue (promises)
+      if (promises.findBy('url', url)) {
+        // get the filterd promise object
+        var filteredPromise = promises.findBy('url', url);
+
+        later = Ember.RSVP.defer();
+
+        filteredPromise.promises.push(later);
+
+        later = later.promise;
+
+      } else { // request is not in the promiseQueue
+
+        later = self.request({
+          url: url,
+          depaginate: opt.depaginate,
+          headers: newHeaders,
+        }).then((result) => {
+          var localPromises = self.get('promiseQueue').findBy('url', url);
```

Comment:
Seems like this should be a map/object with the URL as the key so you don't have to search the array every time.